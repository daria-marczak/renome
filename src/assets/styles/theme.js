export const theme = {
  primary: '#985e5b',
  black: '#000',
  white: '#fff',
  lightGray: '#dadada',
  gray: '#545459',
  font: {
    family: {
      lato: 'Lato, sans-serif',
      montserrat: 'Montserrat, sans-serif',
      playfair: 'Playfair Display, serif',
    },
    size: {
      header: '38px',
      paragraph: '14px',
      jumbotron: '22px',
      paragraphDescription: '11px',
      carouselDescription: '16px',
    },
  },
};
