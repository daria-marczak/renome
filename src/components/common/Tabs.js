import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledTabContainer = styled.div`
  display: flex;
  min-width: 375px;
  justify-self: center;
  border-bottom: 2px solid ${({ theme }) => theme.lightGray};

  @media (min-width: 1200px) {
    min-width: 400px;
    justify-self: center;
  }
`;

const StyledTabLink = styled.div`
  border-bottom: 5px solid
    ${({ theme, isActive }) => (isActive ? theme.primary : 'transparent')};
  flex: 1;
  font-weight: 700;
  text-align: center;
  line-height: 40px;
  text-transform: uppercase;
  transition: border-bottom 0.3s ease-in-out;
  cursor: pointer;

  @media (min-width: 1200px) {
    max-width: 130px;
  }
`;

const Tabs = ({ tabs, activeTab, setTab }) => {
  return (
    <StyledTabContainer>
      {tabs.map(tab => (
        <StyledTabLink
          key={tab}
          onClick={() => setTab(tab)}
          isActive={activeTab === tab}
        >
          {tab}
        </StyledTabLink>
      ))}
    </StyledTabContainer>
  );
};

Tabs.propTypes = {
  tabs: PropTypes.arrayOf(PropTypes.string).isRequired,
  activeTab: PropTypes.string.isRequired,
  setTab: PropTypes.func.isRequired,
};

export default Tabs;
