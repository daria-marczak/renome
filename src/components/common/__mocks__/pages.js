export const pages = [
  'home',
  'about',
  'menu',
  'reservations',
  'blog',
  'shop',
  'contact',
];
