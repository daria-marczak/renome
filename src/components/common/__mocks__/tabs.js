export const tabs = ['Breakfast', 'Dinner', 'Drinks'];
