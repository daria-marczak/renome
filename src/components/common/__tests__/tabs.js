import React from 'react';
import { render } from '@testing-library/react';
import Tabs from '../Tabs';
import theme from '../../../assets/styles/theme';
import * as tabs from '../__mocks__/tabs';

jest.mock('./tabs');
const setTab = jest.fn;

describe('Tabs', () => {
  const { getByTestId, container } = render(
    <Tabs
      theme={theme}
      tabs={tabs.tabs}
      activeTab="Breakfast"
      setTab={setTab}
    />
  );

  it('renders component', () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has active tab with correct text', () => {
    expect(getByTestId('activeTab').innerHTML).toBe('Breakfast');
  });
});
