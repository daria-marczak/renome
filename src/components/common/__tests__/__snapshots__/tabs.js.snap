// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders component 1`] = `
<div
  class="sc-bdVaJa fDZkqc"
  data-testid="tabsComponent"
>
  <div
    class="sc-bwzfXH ihWUwL"
    data-testid="activeTab"
  >
    Breakfast
  </div>
  <div
    class="sc-bwzfXH bjqPYq"
    data-testid=""
  >
    Dinner
  </div>
  <div
    class="sc-bwzfXH bjqPYq"
    data-testid=""
  >
    Drinks
  </div>
</div>
`;

exports[`Tabs renders correctly 1`] = `
ReactShallowRenderer {
  "_context": null,
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useEffect": [Function],
    "useImperativeHandle": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useState": [Function],
  },
  "_element": null,
  "_firstWorkInProgressHook": null,
  "_forcedUpdate": false,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": null,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;
